### PR TITLE
Adds rule to EIP-1 that references to other EIPs must use relative path format and the first reference must be linked.

### PR DESCRIPTION
**Pull request by MicahZoltu**
I have gone through and updated all existing EIPs to match this rule, including EIP-1.

In some cases, people were using markdown citations, I suspect because the long-form was a bit verbose to inline.  Since the relative path is quite short, I moved these to inline but I wouldn't be opposed to putting them back to citation format if that is desired by the authors.

In doing the migration/cleanup, I found some EIP references to EIPs that don't actually exist.  In these cases I tried to excise the reference from the EIP as best I could.

It is worth noting that the Readme actually already had this rule, it just wasn't expressed properly in EIP-1 and the "Citation Format" section of the readme I think caused people a bit of confusion (when citing externally, you should use the citation format).

cc @axic @lightclient @Souptacular @gcolvin.  I propose each of you approve or reject this PR, and if everyone approves we can merge it in without further delay.  If someone rejects, we can use that as a signal that we need further discussion.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
If someone can provide some insight as to why the build is failing that would be much appreciated.  It says that `./eip-1702.md` doesn't exist when linked from `EIPS/eip-615.md`.  1702 shows up on GitHub, and on the EIPs site, and I have verified neither the link nor the filename have any tricky characters in them (e.g., `O` instead of `0`).

**Comment by axic** [comment]
> It is worth noting that the Readme actually already had this rule, it just wasn't expressed properly in EIP-1 and the "Citation Format" section of the readme I think caused people a bit of confusion (when citing externally, you should use the citation format).

The readme is not in sync with EIP-1, and EIP-1 takes precedence. Some people updated the README when updating EIP-1, other did not. There was also some push to drop all this duplication from EIP-1.

I'd have suggested to open a discussion topic on this before going lengths at doing the changes :)

**Comment by MicahZoltu** [comment]
> I'd have suggested to open a discussion topic on this before going lengths at doing the changes :)

Meh, if we don't follow through it isn't a big deal.  Took me maybe 15-30 minutes to get all the right regex find/replaces in.  Also, if we do decide to go with eips website only, it will be very simple to convert this PR into that now that all EIP links are standardized across the repository.

> The readme is not in sync with EIP-1, and EIP-1 takes precedence. Some people updated the README when updating EIP-1, other did not. There was also some push to drop all this duplication from the readme.

I think you already know this, but for those who haven't been following our recent discussions on this topic:
* EIP-1 doesn't explicitly specify how *internal* EIP links should handled
* we have "Citation Format" which says anyone citing an EIP should use the EIPs website
* we have that excerpt from the readme (which is non-normative) that says EIP links should be relative

**Comment by axic** [comment]
There's a lot of info on this topic on the #eips discord channel. There was some reason brought up earlier for not using internal links. I'm not sure if that is still valid, but we should verify those claims (by checking a local build with these changes, etc.) before merging anything like this.

More importantly I think we should clean up the README and the front page on eips.ethereum.org -- both duplicate a lot of content with EIP-1 and in most cases in an inconsistent manner. I think EIP-1 should describe **all the rules**, and perhaps some technicality regarding GitHub and the automerger can remain outside of it, but wouldn't fight against even keeping that within some appendix of EIP-1.

**Comment by MicahZoltu** [comment]
> More importantly I think we should clean up the README and the front page on eips.ethereum.org -- both duplicate a lot of content with EIP-1 and in most cases in an inconsistent manner. I think EIP-1 should describe all the rules, and perhaps some technicality regarding GitHub and the automerger can remain outside of it, but wouldn't fight against even keeping that within some appendix of EIP-1.

I support this endeavor.  However, I think it should be separate from this PR to keep the discussions/blockers/disagreements/etc. separate.

> There was some reason brought up earlier for not using internal links. I'm not sure if that is still valid, but we should verify those claims (by checking a local build with these changes, etc.) before merging anything like this.

According to https://github.com/ethereum/EIPs/pull/1730#issuecomment-459516912, relative linking currently works on the EIPs site (also my personal experience agrees with this, and the HTML-proofer tangentially validates this).  Does anyone here have the infrastructure setup to run this PR locally to further validate?  Maybe @lightclient since you were previously working on some bot stuff?

Also according to that comment, I believe the primary argument for `eips.ethereum.org` hard links was to try to redirect people to the EIPs site and away from GitHub in case they stumble upon GitHub.  My argument against that I believe the intersection of people who land on some random GitHub fork of the EIPs repository and the people who are browsing the EIPs repository by clicking on links is incredibly small and not worth breaking appropriate linking for people working in pull requests, mirrors, forks, etc.

**Comment by MicahZoltu** [comment]
Discussed on call today, will rebase tomorrow and merge unless someone says something.

**Comment by MicahZoltu** [comment]
Ugh, I actually still need help on figuring out why the HTML proofer is failing on that one EIP.  😢

**Comment by axic** [comment]
> Discussed on call today, will rebase tomorrow and merge unless someone says something.

What call do you mean?

**Comment by MicahZoltu** [comment]
EIPIP call.

**Comment by axic** [comment]
> EIPIP call.

Hm, okay.

> There's a lot of info on this topic on the #eips discord channel. There was some reason brought up earlier for not using internal links. I'm not sure if that is still valid, but we should verify those claims (by checking a local build with these changes, etc.) before merging anything like this.

Did anyone verify that everything works with these changes?

**Comment by MicahZoltu** [comment]
Finally fixed that issue with 615.  I didn't realize there were two different links to 1702 in the file and one of them was wrong (I kept checking the right one when investigating).

I also have verified that relative links work on the eips site, though I don't think there was any reason to doubt this seeing as how that is how all links *used* to be, and many links today are currently relative.

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 1 is in state Active, not Draft or Last Call
 - EIP 1013 is in state Final, not Draft or Last Call
 - EIP 1014 is in state Final, not Draft or Last Call
 - EIP 1052 is in state Final, not Draft or Last Call
 - EIP 1108 is in state Final, not Draft or Last Call
 - EIP 1123 is in state Abandoned, not Draft or Last Call
 - EIP 1155 is in state Final, not Draft or Last Call
 - EIP 1193 is in state Final, not Draft or Last Call
 - EIP 1234 is in state Final, not Draft or Last Call
 - EIP 1344 is in state Final, not Draft or Last Call
 - EIP 137 is in state Final, not Draft or Last Call
 - EIP 162 is in state Final, not Draft or Last Call
 - EIP 1679 is in state Final, not Draft or Last Call
 - EIP 1682 is in state Abandoned, not Draft or Last Call
 - EIP 1706 is in state Abandoned, not Draft or Last Call
 - EIP 1820 is in state Final, not Draft or Last Call
 - EIP 1884 is in state Final, not Draft or Last Call
 - EIP 190 is in state Final, not Draft or Last Call
 - Error checking file EIPS/eip-1948.md
 - Error checking file EIPS/eip-1996.md
 - File EIPS/eip-20-token-standard.md is not an EIP
 - Error checking file EIPS/eip-2009.md
 - Error checking file EIPS/eip-2018.md
 - Error checking file EIPS/eip-2019.md
 - Error checking file EIPS/eip-2020.md
 - Error checking file EIPS/eip-2021.md
 - Error checking file EIPS/eip-2098.md
 - Error checking file EIPS/eip-2124.md
 - EIP 2200 is in state Final, not Draft or Last Call
 - Error checking file EIPS/eip-2304.md
 - Error checking file EIPS/eip-2334.md
 - Error checking file EIPS/eip-2335.md
 - Error checking file EIPS/eip-2364.md
 - EIP 2384 is in state Final, not Draft or Last Call
 - EIP 2387 is in state Final, not Draft or Last Call
 - Error checking file EIPS/eip-2477.md
 - EIP 607 is in state Final, not Draft or Last Call
 - EIP 609 is in state Final, not Draft or Last Call
 - Error checking file EIPS/eip-615.md
 - Error checking file EIPS/eip-689.md
 - EIP 695 is in state Final, not Draft or Last Call
 - Error checking file EIPS/eip-712.md
 - EIP 721 is in state Final, not Draft or Last Call
 - EIP 777 is in state Final, not Draft or Last Call
 - EIP 820 is in state Superseded, not Draft or Last Call
 - Error checking file EIPS/eip-831.md
 - EIP 908 is in state Abandoned, not Draft or Last Call
 - Error checking file EIPS/eip-969.md
 - Error checking file EIPS/eip-998.md
 - File ISSUE_TEMPLATE.md is not an EIP
 - File README.md is not an EIP
 - EIP 1015 requires approval from one of (avsa@ethereum.org)
 - EIP 1056 requires approval from one of (pelle.braendgaard@consensys.net, oed@consensys.net)
 - EIP 1066 requires approval from one of (@carchrae, @naumenkogs, @expede)
 - EIP 1077 requires approval from one of (@3esmit, avsa@ethereum.org)
 - EIP 1078 requires approval from one of (avsa@ethereum.org)
 - EIP 1202 requires approval from one of (@yingogobot, @xinbenlv, @evbots)
 - EIP 1261 requires approval from one of (@chaitanyapotti, @pb25193)
 - EIP 1276 requires approval from one of (@eosclassicteam)
 - EIP 1417 requires approval from one of (@chaitanyapotti, @pb25193)
 - EIP 1444 requires approval from one of (@expede, @jenncoop)
 - EIP 1459 requires approval from one of (@fjl, peter@ethereum.org)
 - EIP 1462 requires approval from one of (mk@atlant.io, js@atlant.io)
 - EIP 1474 requires approval from one of (@bitpshr, rekmarks@protonmail.com)
 - EIP 1484 requires approval from one of (@shanehampton, andychorlian@gmail.com, anurag.angara@gmail.com, @noahzinsmeister)
 - EIP 1485 requires approval from one of (cpplover@trustfarm.net, @trustfarm-dev)
 - EIP 1592 requires approval from one of (cyril.lapinte@mtpelerin.com, laurent.aapro@mtpelerin.com)
 - EIP 1613 requires approval from one of (alex@tabookey.com, dror@tabookey.com, @yoav-tabookey)
 - EIP 1710 requires approval from one of (@brunobar79)
 - EIP 173 requires approval from one of (@danfinlay, @mudgen)
 - EIP 1775 requires approval from one of (@danfinlay, @bunjin)
 - EIP 1803 requires approval from one of (@axic)
 - EIP 1812 requires approval from one of (@pelle)
 - EIP 1829 requires approval from one of (recmo@0x.org)
 - EIP 1898 requires approval from one of (@charles-cooper)
 - EIP 1901 requires approval from one of (@shanejonas, @belfordz)
 - EIP 191 requires approval from one of (@arachnid, @holiman)
 - EIP 1921 requires approval from one of (@ctzurcanu, @loredanacirstea)
 - EIP 1922 requires approval from one of (chaitanya.konda@uk.ey.com, @iammichaelconnor, @westlad)
 - EIP 1923 requires approval from one of (chaitanya.konda@uk.ey.com, @iammichaelconnor, @westlad)
 - EIP 1930 requires approval from one of (@wighawag)
 - EIP 1959 requires approval from one of (@wighawag)
 - EIP 1962 requires approval from one of (@shamatar)
 - EIP 1965 requires approval from one of (@wighawag)
 - EIP 1967 requires approval from one of (@spalladino)
 - EIP 1985 requires approval from one of (@chfast, @axic)
 - EIP 2003 requires approval from one of (@chfast, @axic)
 - EIP 2014 requires approval from one of (@axic)
 - EIP 2015 requires approval from one of (@pedrouid)
 - EIP 2025 requires approval from one of (@madeoftin)
 - EIP 2029 requires approval from one of (@alexeyakhunov)
 - EIP 2031 requires approval from one of (@alexeyakhunov)
 - EIP 2035 requires approval from one of (@alexeyakhunov)
 - EIP 2045 requires approval from one of (@cdetrio, @axic)
 - EIP 2046 requires approval from one of (@axic)
 - EIP 2070 requires approval from one of (@axic)
 - EIP 2135 requires approval from one of (@xinbenlv)
 - EIP 2193 requires approval from one of (@ctzurcanu, @loredanacirstea)
 - EIP 2242 requires approval from one of (@adlerjohn)
 - EIP 2255 requires approval from one of (@danfinlay, @rekmarks)
 - EIP 2256 requires approval from one of (@loredanacirstea)
 - EIP 2327 requires approval from one of (@mrchico)
 - EIP 233 requires approval from one of (@axic)
 - EIP 2400 requires approval from one of (@3esmit, @yenda)
 - EIP 2470 requires approval from one of (@3esmit)
 - EIP 2488 requires approval from one of (@axic)
 - EIP 2525 requires approval from one of (@amxx)
 - EIP 2535 requires approval from one of (@mudgen)
 - EIP 2565 requires approval from one of (@simonatsn, @dankrad, @justindrake, @sean-sn, @ineffectualproperty)
 - EIP 2569 requires approval from one of (@zhous, @dgczhh, @whtyfhas)
 - EIP 2583 requires approval from one of (@holiman)
 - EIP 2612 requires approval from one of (@mrchico)
 - EIP 2677 requires approval from one of (@chfast, @axic, @holiman)
 - EIP 2678 requires approval from one of (@gnidan, @chriseth, @iamdefinitelyahuman, @fubuloubu, @pipermerriam, @njgheorghita)
 - EIP 2681 requires approval from one of (@axic)
 - EIP 2733 requires approval from one of (@lightclient)
 - EIP 2746 requires approval from one of (@juanfranblanco, @jaerith)
 - EIP 2926 requires approval from one of (@axic, @s1na)
 - EIP 2929 requires approval from one of (@holiman, @vbuterin)
 - EIP 663 requires approval from one of (@axic)
 - EIP 725 requires approval from one of (@tyleryasaka, @frozeman)
 - EIP 868 requires approval from one of (@fjl)

**Review by chaitanyapotti — APPROVED** [review]
Lgtm

**Comment by MicahZoltu** [comment]
Please ignore the bot mentions here.  This is a cleanup by the editors to fix some inconsistencies, no need for everyone to review.  😄

File: EIPS/eip-1015.md [hunk 1]
```diff
@@ -28,16 +28,16 @@ Moving to PoS has been on the roadmap since day 0 for ethereum, along with a red
...
+[EIP 960](./eip-969.md) proposes a change in algorithm to avoid mining being dominated by ASICS. Counter arguments by Phil Daian argue among others than [resisting economies of scale is futile and there might be specific security advantadges to specialized hardware](https://pdaian.com/blog/anti-asic-forks-considered-harmful/). One of the main arguments for PoW mining, even when it doesn't provide security, it is useful as a fair distribution mechanism, that **PoW allows any person with a computer, internet access and electricity to obtain currency without having to deal with government imposed currency controls**.
```
Comment (axic) [inline EIPS/eip-1015.md:31]: So, is it EIP 960 or EIP-960?

Comment (axic) [reply EIPS/eip-1015.md:31]: And why does it link to eip-969 ?

File: EIPS/eip-1056.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ This ERC describes a standard for creating and updating identities with a limite
 
 ## Motivation
 
-As we have been developing identity systems for the last couple of years at uPort it has become apparent that the cost of identity creation is a large issue. The previous Identity proposal [ERC725](https://eips.ethereum.org/EIPS/eip-725) faces this exact issue. Our requirements when creating this ERC is that identity creation should be free, and should be possible to do in an offline environment (e.g. refugee scenario). However it must also be possible to rotate keys without changing the primary identifier of the identity. The identity system should be fit to use off-chain as well as on-chain.
+As we have been developing identity systems for the last couple of years at uPort it has become apparent that the cost of identity creation is a large issue. The previous Identity proposal [ERC725](./eip-725.md) faces this exact issue. Our requirements when creating this ERC is that identity creation should be free, and should be possible to do in an offline environment (e.g. refugee scenario). However it must also be possible to rotate keys without changing the primary identifier of the identity. The identity system should be fit to use off-chain as well as on-chain.
```
Comment (axic) [inline EIPS/eip-1056.md:22]: And these ERC links don't have a dash.

File: EIPS/eip-1155.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ pragma solidity ^0.5.9;
 
 /**
     @title ERC-1155 Multi Token Standard
-    @dev See https://eips.ethereum.org/EIPS/eip-1155
+    @dev See ./eip-1155.md
```
Comment (axic) [inline EIPS/eip-1155.md:42]: I'm not so sure about this change. I assume this interface is copy pasted somewhere else and then this link becomes useless. Moreso, this is a Natspec definition which will be part of the contract metadata and this link will be useless.

File: EIPS/eip-1462.md [hunk 4]
```diff
@@ -16,15 +16,15 @@ An extension to ERC-20 standard token that provides compliance with securities r
...
 
 ## Motivation
 
-There are several security token standards that have been proposed recently. Examples include [ERC-1400](https://github.com/ethereum/EIPs/issues/1411), also [ERC-1450](https://eips.ethereum.org/EIPS/eip-1450). We have concerns about each of them, mostly because the scope of each of these EIPs contains many project-specific or market-specific details. Since many EIPs are coming from the respective backing companies, they capture many niche requirements that are excessive for a general case.
+There are several security token standards that have been proposed recently. Examples include [ERC-1400](https://github.com/ethereum/EIPs/issues/1411), also [ERC-1450](./eip-1450.md). We have concerns about each of them, mostly because the scope of each of these EIPs contains many project-specific or market-specific details. Since many EIPs are coming from the respective backing companies, they capture many niche requirements that are excessive for a general case.
```
Comment (axic) [inline EIPS/eip-1462.md:25]: > `[ERC-1400](https://github.com/ethereum/EIPs/issues/1411)`

Is it 1400 or 1411?

File: EIPS/eip-1462.md [hunk 5]
```diff
@@ -16,15 +16,15 @@ An extension to ERC-20 standard token that provides compliance with securities r
...
+For instance, ERC-1411 uses dependency on [ERC-1410](https://github.com/ethereum/eips/issues/1410) but it falls out of the "security tokens" scope. Also its dependency on [ERC-777](./eip-777.md) will block the adoption for a quite period of time before ERC-777 is finalized, but the integration guidelines for existing ERC-20 workflows are not described in that EIP, yet. Another attempt to make a much simpler base standard [ERC-1404](https://github.com/ethereum/EIPs/issues/1404) is missing a few important points, specifically it doesn't provide enough granularity to distinguish between different ERC-20 transfer functions such as `transfer` and `transferFrom`. It also doesn't provide a way to bind legal documentation to the issued tokens.
```
Comment (axic) [inline EIPS/eip-1462.md:27]: And then here it calls "ERC-1411" which is not linked.

File: EIPS/eip-1682.md [hunk 6]
```diff
@@ -32,13 +32,7 @@ upper bound on the size of the Ethereum state.
 Adding these new rules changes fundamental guarantees of the system and requires a hard
 fork. Users of Ethereum already pay for the creation and modification of accounts and
 their storage entries. Under the rules introduced in this EIP, users must also pay to keep
-accounts accessible. A similar rent scheme was proposed in [EIP-103] but rejected
```
Comment (axic) [inline EIPS/eip-1682.md:35]: These very early EIPs were handled differently. They had some random number picked, and were kept in an issue with a different number.

I think removing this on the bases that it is an "invalid link" is not good enough.

File: EIPS/eip-1820.md [hunk 7]
```diff
@@ -102,7 +102,7 @@ interface ERC1820ImplementerInterface {
 /// @title ERC1820 Pseudo-introspection Registry Contract
 /// @author Jordi Baylina and Jacques Dafflon
 /// @notice This contract is the official implementation of the ERC1820 Registry.
-/// @notice For more details, see https://eips.ethereum.org/EIPS/eip-1820
+/// @notice For more details, see ./eip-1820.md
```
Comment (axic) [inline EIPS/eip-1820.md:105]: Again, natspec change.

File: EIPS/eip-1820.md [hunk 8]
```diff
@@ -651,7 +651,7 @@ The contract has the address above for every chain on which it is deployed.
...
...;\n    }\n\n    /// @dev Make a call on a contract without throwing if the function does not exist.\n    function noThrowCall(address _contract, bytes4 _interfaceId)\n        internal view returns (uint256 success, uint256 result)\n    {\n        bytes4 erc165ID = ERC165ID;\n\n        assembly {\n            let x := mload(0x40)               // Find empty storage location using \"free memory pointer\"\n            mstore(x, erc165ID)                // Place signature at beginning of empty storage\n            mstore(add(x, 0x04), _interfaceId) // Place first argument directly next to signature\n\n            success := staticcall(\n                30000,                         // 30k gas\n                _contract,                     // To addr\n                x,                             // Inputs are stored at location x\n                0x24,                          // Inputs are 36 (4 + 32) bytes long\n                x,                             // Store output over input (saves space)\n                0x20                           // Outputs are 32 bytes long\n            )\n\n            result := mload(x)                 // Load the result\n        }\n    }\n}\n",
```
Comment (axic) [inline EIPS/eip-1820.md:654]: This changes the source, renders the hash below invalid, and therefore breaks this EIP.

**Comment by MicahZoltu** [comment]
@axic convinced me to redo this PR so it didn't touch as much all at once.  Will change strategy to:
1. Normalize all [EIP?###] to [EIP-###]
2. Normalize all [ERC?###] to [ERC-###]
3. Convert all [EIP-###](?) to [EIP-###](./eip-###.md)
4. Convert all [ERC-###](?) to [ERC-###](./eip-###.md)
5. Wait for bot to run.
6. Revert everything the bot complains about.
7. Wait for bot to pass.
8. Get approval from Axic/Lightclient.
9. Merge.

**Comment by BelfordZ** [comment]
👀 👍

**Comment by 3esmit** [comment]
Approved (EIP 2400 and EIP 2470).

**Comment by MicahZoltu** [comment]
> looks valid though I'm not sure I understand the motivation behind wanting only relative links. This makes the EIPs no longer able to be consumed in a stand-alone fashion.

I believe the opposite is true?  When absolute links are used the document breaks when you are working on a fork, or hosting on a mirror, or browsing via IPFS, or local filesystem while offline, etc..  If you use relative links, then no matter how you are browsing the EIP repository, you'll be able to access everything.

**Review by lightclient — APPROVED** [review]
In principle, I'm on board with this PR. I have not had a chance to go through it line-by-line though.

**Comment by MicahZoltu** [comment]
We can always fix mistakes afterward if necessary, so "in principle" is good enough for me.  😀

**Comment by BelfordZ** [comment]
> > looks valid though I'm not sure I understand the motivation behind wanting only relative links. This makes the EIPs no longer able to be consumed in a stand-alone fashion.
> 
> I believe the opposite is true? When absolute links are used the document breaks when you are working on a fork, or hosting on a mirror, or browsing via IPFS, or local filesystem while offline, etc.. If you use relative links, then no matter how you are browsing the EIP repository, you'll be able to access everything.

Perhaps I was not clear enough by standalone. If you send the contents of an EIP in an email, none of the links will work. If you use anything other than a web server which is also serving all the other eips, then nothing works.

The opposite certainly is not true.

**Comment by lightclient** [comment]
@BelfordZ happy to be proven otherwise, but I think the category of users you're describing is very small. Also, there is a lurking variable in your issue. I think the goal should be that an EIP should be understandable without access to the external resources (assuming the reader has the correct context), so if that's the case then the link format shouldn't matter too much to those users.

**Comment by MicahZoltu** [comment]
I have been unable to get a review from any editors except lightclient, so I'm going to merge this as is.  We generally green lit this change in an EIPIP meeting, but I was hoping to get some eyes to make sure I didn't screw up the migration.  However, since everyone seems to be too busy I'm going to make an executive decision and merge this as is.

**Pull request merged**